### ISSUE TITLE
[Suggestion] Restructure ReactXP as a lerna monorepo to facilitate internal projects

### ISSUE THREAD

**Issue by luisnaranjo733** [Microsoft/reactxp#844]
I think maintaining ReactXP would be easier if it were structured as a[ lerna monorepo.](https://lernajs.io/)

Example scenario today:
* You make a breaking change to the XP source
* You want to add a test case to test your test project to test this change, but this project uses a published version of XP, so you can only test by manually building XP and copying the source over to node_modules.

Same scenario with a lerna monorepo
* You make a breaking change
* Changes will be automatically available in node_modules of the internal project because ```lerna bootstrap``` symlinks node_modules for you

The one caveat I should mention is that RN metro bundler doesn't like symlinks in node_modules. You may have the same problem when testing RN. There is probably a way to configure lerna so that it copies instead of symlinks when running ```lerna bootstrap```

**Comment by erictraut** [comment]
I don't know if lerna is the answer, but I agree that this is a problem. Another potential solution is to have the RXPTest sample directly import from the core ReactXP sources rather than importing the published version.

**Comment by fbartho** [comment]
I'm motivated to get some sort of improved tooling for testing, but I think we can get that done with simpler tools than lerna.

Was there anything further you wanted to request in this ticket @luisnaranjo733 ?

See also #1233

**Comment by luisnaranjo733** [comment]
Nope! My intent was to outline the problem, not to be prescriptive about the solution. Thanks!
